gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageHttpRequestFormatter.java

```diff
@@ -56,20 +65,17 @@
   }
 
   @Override
-  public Map<String, String> getPathParams(ApiMessage apiMessage, Set<String> paramNames) {
-    Map<String, String> pathParams = new HashMap<>();
-    Map<String, List<String>> pathParamMap = apiMessage.populateFieldsInMap(paramNames);
-    Iterator<Map.Entry<String, List<String>>> iterator = pathParamMap.entrySet().iterator();
-    while (iterator.hasNext()) {
-      Map.Entry<String, List<String>> pair = iterator.next();
-      pathParams.put(pair.getKey(), pair.getValue().get(0));
-    }
-    return pathParams;
+  public Map<String, String> getPathParams(T apiMessage, String resourceNameField) {
+    Map<String, List<String>> pathParamMap =
+        apiMessage.populateFieldsInMap(Collections.singleton(resourceNameField));
+
+    String resourceNamePath = pathParamMap.get(resourceNameField).get(0);
+    return resourceNameInstance.parseFrom(resourceNamePath).getFieldValues();
```

Comment:
I'm a little confused about why we need to construct an instance of the resource name in the process of executing a user's request.

Reply:
The ResourceName instance can get a Map<String, String> of field names to values.
This String->String map is used to instantiate the endpoint relative path, e.g. for method `compute.insert.addresses` the template is 
`"projects/{projects}/regions/{region}/addresses"`

The path template for, say, RegionName is only `"projects/{project}/regions/{region}"`, so the generated message's `region` formatted String field lacks the trailing `"/addresses"` substring to create the endpoint relative path for the `compute.insert.addresses` method.

Therefore, the message's String `region` has to be deserialized to a RegionName, the Map<String, String> of applicable fields (`project` and `region`) must be extracted from the RegionName, and then formatted with the endpoint path template to create the relative endpoint path.